Sources/SwiftGenKit/Parsers/StringsParser.swift

```diff
@@ -37,23 +37,58 @@ public final class StringsParser: Parser {
 
   // Localizable.strings files are generally UTF16, not UTF8!
   public func parse(path: Path) throws {
-    let name = path.lastComponentWithoutExtension
+    if path.isDirectory {
 
-    guard tables[name] == nil else {
-      throw StringsParserError.duplicateTable(name: name)
-    }
-    guard let data = try? path.read() else {
-      throw StringsParserError.failureOnLoading(path: path.string)
-    }
+      if try hasMultipleLprojSubdirsWithDifferentLanguages(path: path) {
+        warningHandler?("Multiple .lproj directories found with different languages." +
+          "Duplicate keys will be parsed multiple times.", #file, #line)
+      }
+
+      try _ = path.children().map { childPath in
+        if childPath.isDirectory || isStringsFile(name: childPath.lastComponent) {
+          try parse(path: childPath)
+        }
+      }
+    } else {
+      let name = path.lastComponentWithoutExtension
+
+      guard tables[name] == nil else {
+        throw StringsParserError.duplicateTable(name: name)
+      }
+      guard let data = try? path.read() else {
+        throw StringsParserError.failureOnLoading(path: path.string)
+      }
 
-    let plist = try PropertyListSerialization.propertyList(from: data, format: nil)
-    guard let dict = plist as? [String: String] else {
-      throw StringsParserError.invalidFormat
+      let plist = try PropertyListSerialization.propertyList(from: data, format: nil)
+      guard let dict = plist as? [String: String] else {
+        throw StringsParserError.invalidFormat
+      }
+
+      tables[name] = try dict.map { key, translation in
+        try Entry(key: key, translation: translation)
+      }
     }
+  }
 
-    tables[name] = try dict.map { key, translation in
-      try Entry(key: key, translation: translation)
+  private func hasMultipleLprojSubdirsWithDifferentLanguages(path: Path) throws -> Bool {
+    var lprojSubdir: String?
+    for childPath in try path.children() {
+      if childPath.lastComponent.hasSuffix(".lproj") {
```

Comment:
`childPath` is already a `Path` — which is a type coming from PathKit to ease manipulation of file paths.
So instead of using `lastComponent` to get a `String` and test for `hasSuffix`, it would be simpler and better to directly use `childPath.pathExtension` for the test :wink:

Also, I try to avoid magic strings. It would be a good idea to create constants for those path extensions we test, and check everywhere in the code if we use `"lproj"` and `"strings"` to replace them with those constants

```swift
enum FileExtensions {
  static let lproj = "lproj"
  static let strings = "strings"
}
```